x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/use_fetch_analytics_map_data.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { uniqWith, isEqual } from 'lodash';
+import cytoscape from 'cytoscape';
+import { ml } from '../../../services/ml_api_service';
+import { JOB_MAP_NODE_TYPES } from '../../../../../common/constants/data_frame_analytics';
+
+interface GetDataObjectParameter {
+  analyticsId?: string;
+  id?: string;
+  modelId?: string;
+  type?: string;
+}
+
+export const useFetchAnalyticsMapData = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [elements, setElements] = useState<cytoscape.ElementDefinition[]>([]);
+  const [nodeDetails, setNodeDetails] = useState<Record<string, any>>({});
+  const [error, setError] = useState<any>();
+  const [message, setMessage] = useState<string | undefined>();
+
+  const getData = async (idToUse: string, treatAsRoot: boolean, type?: string) => {
+    setIsLoading(true);
+    // Pass in treatAsRoot flag - endpoint will take job or index to grab jobs created from it
+    const analyticsMap: any = await ml.dataFrameAnalytics.getDataFrameAnalyticsMap(
```

Comment:
please defined an interface for `analyticsMap `

Reply:
Added in 24e03b4ac632921bafc7dee7d4cec34c9b22104c

Reply:
https://github.com/elastic/kibana/pull/84299#discussion_r535938301